src/da/da-helper.js

```diff
@@ -0,0 +1,560 @@
+  const { path: pathDep } = dependencies;
+  const assetMapping = new Map();
+  
+  // Extract page name from pagePath to create shadow folder
+  const pageName = pagePath ? pathDep.basename(pagePath, pathDep.extname(pagePath)) : '';
+  const shadowFolder = pageName ? `.${pageName}` : '';
+  
+  assetUrls.forEach(url => {
+    try {
+      // Parse the URL to extract the filename
+      const urlObj = new URL(url);
+      const filename = urlObj.pathname.split('/').pop();
+      const relativePath = shadowFolder ? `/${shadowFolder}/${filename}` : `/${filename}`;
+      assetMapping.set(url, relativePath);
+    } catch (error) {
+      // If URL parsing fails, assume it's already a relative path
+      const filename = url.split('/').pop();
+      const relativePath = shadowFolder ? `/${shadowFolder}/${filename}` : `/${filename}`;
+      assetMapping.set(url, relativePath);
+    }
+  });
+  
+  return assetMapping;
+}
+
+/**
+ * Save updated HTML content back to files
+ * @param {Array<{filePath: string, updatedContent: string}>} processedPages - Array of processed page results
+ * @param {Object} dependencies - Dependencies for testing (optional)
+ * @return {Promise<void>} Promise that resolves when all files are saved
+ */
+export async function saveUpdatedPages(processedPages, dependencies = defaultDependencies) {
+  const { fs: fsDep, chalk: chalkDep } = dependencies;
+  
+  for (const page of processedPages) {
+    if (page.error) {
+      console.log(chalkDep.red(`Skipping ${page.filePath} due to error: ${page.error}`));
+      continue;
+    }
+    
+    try {
+      // Save the updated content back to the original file
+      fsDep.writeFileSync(page.filePath, page.updatedContent, 'utf-8');
+      console.log(chalkDep.green(`Updated page: ${page.filePath}`));
+      
+    } catch (error) {
+      console.error(chalkDep.red(`Error saving updated page ${page.filePath}:`, error.message));
+    }
+  }
+}
+
+// Export updateHrefsInHTML for testing
+export { updateHrefsInHTML };
+
+/**
+ * Convert asset URLs to a simplified mapping for direct download to shadow folder
+ * @param {Array<string>} matchingHrefs - Array of matching asset URLs
+ * @param {string} fullShadowPath - The full shadow folder path
+ * @param {Object} dependencies - Dependencies for testing (optional)
+ * @return {Map<string, string>} Asset mapping for download
+ */
+function createSimplifiedAssetMapping(matchingHrefs, fullShadowPath) {
+  const simplifiedAssetMapping = new Map();
+  
+  matchingHrefs.forEach(url => {
+    try {
+      // Parse the URL to extract the filename
+      const urlObj = new URL(url);
+      const filename = urlObj.pathname.split('/').pop();
+      simplifiedAssetMapping.set(url, `/${fullShadowPath}/${filename}`);
+    } catch (error) {
+      // If URL parsing fails, assume it's already a relative path
+      const filename = url.split('/').pop();
+      simplifiedAssetMapping.set(url, `/${fullShadowPath}/${filename}`);
+    }
+  });
```

Comment:
Considering that we can have urls with query selectors (Eg. `https://example.com/assets/image.jpg?v=123`):
```suggestion
  const getFilename = (url) => {
    try {
      return new URL(url).pathname.split('/').pop();
    } catch {
      return url.split('?')[0].split('#')[0].split('/').pop();
    }
  };

  return new Map(
    matchingHrefs.map(url => [url, `/${fullShadowPath}/${getFilename(url)}`])
  );
```